src/utils/resources/vmi/utils/cloud-init.ts

```diff
@@ -1,23 +1,47 @@
+import { load } from 'js-yaml';
+
 import { V1VirtualMachineInstance } from '@kubevirt-ui/kubevirt-api/kubevirt';
+import { CLOUD_INIT_MISSING_USERNAME } from '@kubevirt-utils/components/Consoles/utils/constants';
+
+type CloudinitUserDataObject = {
+  user?: string;
+  password?: string;
+  passwd?: { users: { name: string }[] };
+};
 
 export const getCloudInitCredentials = (
   vmi: V1VirtualMachineInstance,
-): { user: string; password: string } => {
+): { users: { name?: string; password?: string }[] } => {
```

Comment:
once you add CLOUD_INIT_MISSING_USERNAME to line 38, the name isn't optional but guaranteed, which is much better,  and can remove the?

Reply:
I don't think so. if we end up [here](https://github.com/kubevirt-ui/kubevirt-plugin/pull/807/files#diff-3876c1f1ef780f9cb2697888d5543d57d66c5f3ea540bfccb3d9c2e70d5f2a16R35) we don't have it guaranteed, and I also think we don't want, because as I've mentioned in some other comment: `CLOUD_INIT_MISSING_USERNAME` was originally used for the case when the username was missing while the password was present - which is not the case for the `passwd` object.